### ISSUE TITLE
untrack does not restore listener when callback throws an exception

### ISSUE THREAD

**Issue by LiQuidProQuo** [solidjs/solid#1323]
### Describe the bug

```js
  createEffect(() => {
    try {
      untrack(() => {
        throw new Error();
      });
    } catch (e) {}
    console.log(count()); // this will not be reactive, listener is still null internally
  });
```

https://github.com/solidjs/solid/blob/454ed51f139219630a80d60570ed455a9b0fdc41/packages/solid/src/reactive/signal.ts#L807-L816

### Your Example Website or App

https://playground.solidjs.com/?hash=1765991341&version=1.4.1

### Steps to Reproduce the Bug or Issue

see repro

### Expected behavior

untrack should also restore listener when throwing.

### Platform

playground

reported by: fabiospampinato

**Comment by ryansolid** [comment]
Yeah we should atleast try/finally any end user provided code.

**Comment by ryansolid** [comment]
Fixed by #1335
